site/src/main/java/uk/nhs/digital/ps/beans/Dataset.java

```diff
@@ -89,10 +89,10 @@ public Publication getParentPublication() {
         return HippoBeanHelper.getParentPublication(this);
     }
 
-    public List<List<String>> getTaxonomyList() {
+    public String getFullTaxonomyList() {
         assertPropertyPermitted(PropertyKeys.TAXONOMY);
 
-        return HippoBeanHelper.getTaxonomyList(getKeys());
+        return String.join(",", HippoBeanHelper.getFullTaxonomyList(this));
```

Comment:
While the Dataset bean is a model for view, emitting a comma-separated string from here feels like tying the model a bit too closely to just one form of presentation. Should the same list of taxonomy values be needed to be displayed using some more elaborate HTML structure at some point, we'd need either to add a new method here, laboriously break the comma-separated string into a list in the template or re-work this method to return a list once again.

Could the method continue to return a list but delegate formatting to Freemarker? I haven't tried it, but [join](https://freemarker.apache.org/docs/ref_builtins_sequence.html#ref_builtin_join) built-in looks like a promising candidate.

Reply:
Good suggestion, have updated